Allow secrets in token of GitHubPullrequestPoller

alexreinking opened the pull request:
What it says in the title. I haven't updated the unit tests because I'm not very familiar with the codebase yet. I have some time to try on my own, but would appreciate any pointers.

Fixes #4921

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

p12tic commented:
@alexreinking : Thanks a lot, this looks good.

As for tests, the tests for this class live in master/buildbot/test/unit/changes/test_github.py. Secrets provider can be setup like this (from master/buildbot/test/unit/www/test_oauth.py):

```
        fake_storage_service = FakeSecretStorage()
        fake_storage_service.reconfigService(secretdict={"client-id": "secretClientId",
                                                         "client-secret": "secretClientSecret"})
        secret_service = SecretManager()
        secret_service.services = [fake_storage_service]
        yield secret_service.setServiceParent(self._master)
```

Thanks!

p12tic commented on master/buildbot/newsfragments/githubpullrequestpoller-secrets.bugfix:
```diff
@@ -0,0 +1 @@
+``GitHubPullrequestPoller`` now supports secrets in its ``token`` argument
```
Nitpick: append ```(:issue:`4921`) ``` to mention the bug number.

alexreinking replied:
Done!

alexreinking commented:
@p12tic - that snippet didn't work for me unless I moved the `reconfigService` call to the end. Looks like both `secret_service.setServiceParent` and `self.master.startService` try to configure the `FakeSecretStorage` service and wind up overwriting the secretdict.

Either way, I added a test by running the same steps as in SimplePR, but with the token set to `Secret('token')`. Let me know if I need to do anything else! I also squashed this down to one commit since my fiddling isn't interesting history-wise.

p12tic reviewed (approved):
Looks great, thanks!

alexreinking commented:
You're welcome! Thank you for helping me with, reviewing, and ultimately accepting my PR!
